### PR TITLE
Use Hostname settings, avoid mounting tmpfs over volumes with mounts

### PR DESCRIPTION
**Pull request by nalind**
Go ahead and default to a hostname from the source image if it included one, and avoid mounting a tmpfs at a volume location if the list of mounts already includes an entry that mounts something else there.

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -185,6 +182,19 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 		}
 	}
 	spec.Mounts = append(mounts, boundMounts...)
+	for _, volume := range b.Volumes() {
+		override := false
+		for _, mount := range spec.Mounts {
+			if mount.Destination == volume {
+				// Already mounting something there, no need for a tmpfs.
+				override = true
+			}
```
Comment (rhatdan) [inline run.go]: Should break here.

Comment (nalind) [inline run.go]: I'm going to have to restructure this to get CI to accept it.

**Comment by nalind** [comment]
Okay, the logic that sets the list of mounts in the runtime spec is now its own function.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+  	1eb1736

**Comment by rhatdan** [comment]
@rh-atomic-bot r+  9e94205

**Pull request closed without merging**